src/main/java/com/datapipe/jenkins/vault/credentials/AbstractAuthenticatingVaultTokenCredential.java

```diff
@@ -14,7 +14,8 @@
  * authentication token. This credential type can explicitly configure the namespace which
  * the authentication method is mounted.
  */
-public abstract class AbstractAuthenticatingVaultTokenCredential extends AbstractVaultTokenCredential {
+public abstract class AbstractAuthenticatingVaultTokenCredential extends
+    AbstractVaultTokenCredentialWithExpiration {
```

Comment:
I need to checkout the code locally to understand the double abstract class, it seems counter intuitive.